google/resource_compute_instance.go

```diff
@@ -1120,6 +1125,10 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 					Type:  "ONE_TO_ONE_NAT",
 					NatIP: d.Get(acPrefix + ".nat_ip").(string),
 				}
+				if ptr, ok := d.GetOk(acPrefix + ".public_ptr_domain_name"); ok {
+					ac.SetPublicPtr = true
```

Comment:
Do you need to set `SetPublicPtr` explicitely to true to turn it off if some was using `public_ptr_domain_name` and now removes it from their config?

Reply:
I don't really know why the API requires that `SetPublicPtr` be true *and* `PublicPtrDomainName` have the value, rather than just doing a null check on the latter, but the docs are clear about that.  Creating the new `AccessConfig` means that it won't be set unless that's set in the current config - we do a full replace.

Reply:
We got bit a few times by that when in order to disable a feature in update, we had to use ForceSend false and nil values otherwise, the server would keep the value it currently had.

Reply:
Makes sense!  I don't think it can happen here at all since we do a full replace.  Let me add that to the test to be 100% sure.